go/chat/storage/storage.go

```diff
@@ -71,8 +77,68 @@ func decode(data []byte, res interface{}) error {
 	return err
 }
 
-func simpleDone(msgs *[]chat1.MessageUnboxed, num int) bool {
-	return len(*msgs) >= num
+type simpleResultCollector struct {
```

Comment:
I think this could be `type simpleResultCollector []chat1.MessageUnboxed`, then in its constructor, initialize it as `make(simpleResultCollector, 0, num)`; the `done()` could be `len(s) == cap(s)`. The advantage is less state and less space. What do you think?